src/app/app.component.html

```diff
@@ -57,26 +57,26 @@ <h1>Welcome {{ username }}</h1>
     </div>
     <div id="layout-sidebar">
         <p-scrollPanel [style]="{width: '100%', height: '900px'}" styleClass="sidebar-menu-container">
-        
+
         <ul class="sidebar-menu">
             <li *ngFor="let menuItem of menuItems">
-                <a *ngIf="!menuItem.group" href="#" [joyrideStep]="menuItem.joyrideStep" [text]="menuItem.text" stepPosition="right" [routerLink]="[menuItem.routerLink]" [ngClass]="{'active-item':menuItem==activeItem}" (click)="menuItemClick($event, menuItem)">
+                <a *ngIf="!menuItem.group" href="#" [joyrideStep]="menuItem.joyrideStep" [text]="menuItem.text" stepPosition="right" [routerLink]="[menuItem.routerLink]" [ngClass]="{'active-item':menuItem.routerLink==activeItem.routerLink}" (click)="menuItemClick($event, menuItem)">
                     <h1>{{menuItem.title}}</h1>
                     <p>{{menuItem.description}}</p>
-                </a> 
+                </a>
                 <p-panel styleClass="menuGroup" *ngIf="menuItem.group" [joyrideStep]="menuItem.joyrideStep" [text]="menuItem.text" stepPosition="right" [toggleable]="true" [collapsed]="true" expandIcon="fa-chevron-down" collapseIcon="fa-chevron-up">
                     <p-header>
                         <h1>{{menuItem.title}}</h1>
                         <p>{{menuItem.description}}</p>
                     </p-header>
                     <ul class="menuGroup-list">
                         <li class="menuGroup-child" *ngFor="let child of menuItem.children">
-                            <a *ngIf="!child.is_external_link" href="#" [routerLink]="[child.routerLink]" [ngClass]="{'active-item':child==activeItem}" (click)="menuItemClick($event, child)">
+                            <a *ngIf="!child.is_external_link" href="#" [routerLink]="[child.routerLink]" [ngClass]="{'active-item':child.routerLink==activeItem.routerLink}" (click)="menuItemClick($event, child)">
                                 <h1>
                                     {{child.title}}
                                 </h1>
                             </a>
-                            <a *ngIf="child.is_external_link" [href]="[child.link]" target="_blank" [ngClass]="{'active-item':child==activeItem}">
+                            <a *ngIf="child.is_external_link" [href]="[child.link]" target="_blank" [ngClass]="{'active-item':child.routerLink==activeItem.routerLink}">
```

Comment:
Is this comparison needed? The external link opens in another tab. Will the menu item get highlighted in this case?